x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/permission/ClusterPermissionTests.java

```diff
@@ -0,0 +1,269 @@
+
+        assertThat(clusterPermission.check("cluster:admin/xpack/security/token/invalidate", mockTransportRequest), is(true));
+        assertThat(clusterPermission.check("cluster:admin/ilm/stop", mockTransportRequest), is(true));
+        assertThat(clusterPermission.check("cluster:admin/xpack/security/privilege/get", mockTransportRequest), is(true));
+        assertThat(clusterPermission.check("cluster:admin/snapshot/status", mockTransportRequest), is(false));
+    }
+
+    public void testClusterPermissionCheckWithEmptyActionPatterns() {
+        final ClusterPermission.Builder builder = ClusterPermission.builder();
+        builder.add(cpThatDoesNothing, Set.of(), Set.of());
+        final ClusterPermission clusterPermission = builder.build();
+        assertThat(clusterPermission.check("cluster:admin/ilm/start", mockTransportRequest), is(false));
+        assertThat(clusterPermission.check("cluster:admin/xpack/security/token/invalidate", mockTransportRequest), is(false));
+    }
+
+    public void testClusterPermissionCheckWithExcludeOnlyActionPatterns() {
+        final ClusterPermission.Builder builder = ClusterPermission.builder();
+        builder.add(cpThatDoesNothing, Set.of(), Set.of("cluster:some/thing/to/exclude"));
+        final ClusterPermission clusterPermission = builder.build();
+        assertThat(clusterPermission.check("cluster:admin/ilm/start", mockTransportRequest), is(false));
+        assertThat(clusterPermission.check("cluster:admin/xpack/security/token/invalidate", mockTransportRequest), is(false));
+    }
+
+    public void testClusterPermissionCheckWithActionPatterns() {
+        final ClusterPermission.Builder builder = ClusterPermission.builder();
+        builder.add(cpThatDoesNothing, Set.of("cluster:admin/*"), Set.of("cluster:admin/ilm/*"));
+        final ClusterPermission clusterPermission = builder.build();
+        assertThat(clusterPermission.check("cluster:admin/ilm/start", mockTransportRequest), is(false));
+        assertThat(clusterPermission.check("cluster:admin/xpack/security/token/invalidate", mockTransportRequest), is(true));
+    }
+
+    public void testClusterPermissionCheckWithActionPatternsAndNoExludePatterns() {
+        final ClusterPermission.Builder builder = ClusterPermission.builder();
+        builder.add(cpThatDoesNothing, Set.of("cluster:admin/*"), Set.of());
+        final ClusterPermission clusterPermission = builder.build();
+        assertThat(clusterPermission.check("cluster:admin/ilm/start", mockTransportRequest), is(true));
+        assertThat(clusterPermission.check("cluster:admin/xpack/security/token/invalidate", mockTransportRequest), is(true));
+    }
+
+    public void testNoneClusterPermissionIsImpliedByAny() {
+        {
+            final ClusterPermission otherClusterPermission = ClusterPermission.builder().build();
+            assertThat(ClusterPermission.NONE.implies(otherClusterPermission), is(true));
+        }
```

Comment:
This artibrary scoping implies that this should just be 2 different test methods.